driver/src/main/scala/api/commands/writeresult.scala

```diff
@@ -32,8 +32,7 @@ sealed trait WriteResult {
 object WriteResult {
   def lastError(result: WriteResult): Option[LastError] = result match {
```

Comment:
> Actually it won't break the current API semantic, as currently a ok with some write errors happens as a Future.successful

Only in the `ordered = false` case. On `ordered = true`we get a failed future with a `GenericDriverException` (or a `LastError` with this PR)